test/unit/commands/start-spec.js

```diff
@@ -1,32 +1,47 @@
 'use strict';
-const expect = require('chai').expect;
+const {expect} = require('chai');
 const sinon = require('sinon');
 const proxyquire = require('proxyquire').noCallThru();
 
 const modulePath = '../../../lib/commands/start';
-const StartCommand = require(modulePath);
 
 describe('Unit: Commands > Start', function () {
     describe('run', function () {
-        let myInstance, mySystem;
+        let myInstance, mySystem, StartCommand;
 
         beforeEach(function () {
+            mySystem = {environment: 'production'};
             myInstance = {
                 checkEnvironment: () => true,
                 running: () => Promise.resolve(false),
                 config: {get: sinon.stub()}
             };
 
-            mySystem = {
-                getInstance: () => myInstance,
-                environment: 'production'
-            };
+            StartCommand = proxyquire(modulePath, {
+                './doctor': {doctorCommand: true},
+                '../utils/get-instance': () => myInstance
+            });
         });
 
         afterEach(function () {
             sinon.restore();
         });
 
+        it('uses getInstance to get proper instance', function () {
+            StartCommand = proxyquire(modulePath, {
+                '../utils/get-instance': sinon.stub().throws(new Error('YES_IT_DOES'))
+            });
+
+            const stop = new StartCommand();
+
+            try {
+                stop.run({}).then(() => {throw new Error('Expected promise rejection')});
```

Comment:
Any suggestion for avoiding the Promise error?

Reply:
for now, maybe wrap the `stop.run` call in a `Promise.resolve()`? that's how it's handled in the base run method I think

Reply:
What promise error are you talking about exactly? I'm not seeing one in the tests log (or when I run it locally)?

Reply:
Oops, I should have been clearer!

Reply:
when would the invalid instance error get bypassed?

Reply:
Under normal circumstances, it shouldn't - I think I'm just overthinking it 🤕

Reply:
Ah ok - no problem 😄 These kind of issues should _hopefully_ be resolved once we can use async/await (i.e. when node 6 support is dropped).